env0/resource_team.go

```diff
@@ -0,0 +1,151 @@
+package env0
+
+import (
+	"context"
+	"errors"
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+)
+
+func resourceTeam() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTeamCreate,
+		ReadContext:   resourceTeamRead,
+		UpdateContext: resourceTeamUpdate,
+		DeleteContext: resourceTeamDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceTeamImport},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of the team",
+				Required:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Description for the team",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceTeamCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+	payload := client.TeamCreatePayload{
+		Name: d.Get("name").(string),
+	}
+	if description, ok := d.GetOk("description"); ok {
+		payload.Description = description.(string)
+	}
+
+	team, err := apiClient.TeamCreate(payload)
+	if err != nil {
+		return diag.Errorf("could not create team: %v", err)
+	}
+
+	d.SetId(team.Id)
+
+	return nil
+}
+
+func resourceTeamRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	team, err := apiClient.Team(d.Id())
+	if err != nil {
+		return diag.Errorf("could not get team: %v", err)
+	}
+
+	d.Set("name", team.Name)
+	d.Set("description", team.Description)
+
+	return nil
+}
+
+func resourceTeamUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	payload := client.TeamUpdatePayload{
+		Name: d.Get("name").(string),
+	}
+	if description, ok := d.GetOk("description"); ok {
+		payload.Description = description.(string)
+	}
+
+	_, err := apiClient.TeamUpdate(d.Id(), payload)
+	if err != nil {
+		return diag.Errorf("could not update team: %v", err)
+	}
+
+	return nil
+}
+
+func resourceTeamDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	apiClient := meta.(client.ApiClientInterface)
+
+	err := apiClient.TeamDelete(d.Id())
+	if err != nil {
+		return diag.Errorf("could not delete team: %v", err)
+	}
+	return nil
+}
+
+func resourceTeamImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	id := d.Id()
+	var getErr diag.Diagnostics
```

Comment:
why you need this instead of returning `nil, err`?
i mean if `uuidErr` is nil then we don't need to make another `if` why not just return? same for the `else`

Reply:
I don't understand what you mean. 🤙🏻